edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.DiagnosticsComponent/Metric.cs

```diff
@@ -0,0 +1,35 @@
+// Copyright (c) Microsoft. All rights reserved.
+
+namespace Microsoft.Azure.Devices.Edge.Agent.DiagnosticsComponent
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Text;
+
+    public class Metric
+    {
+        public DateTime TimeGeneratedUtc { get; }
+        public string Name { get; }
+        public double Value { get; }
+        public string Tags { get; }
+
+        public Metric(DateTime timeGeneratedUtc, string name, double value, string tags)
+        {
+            this.TimeGeneratedUtc = timeGeneratedUtc;
+            this.Name = name;
+            this.Value = value;
+            this.Tags = tags;
+        }
+
+        public int GetValuelessHash()
```

Comment:
what is this method used for? why upgraded to .net standard 2.1? why not implement IEquatable and override Equals and GetHashCode methods?

Reply:
This is used for checking similarity, not equality. It is used to group metrics by name and tags. The net standard 2.1 only refers to using ```HashCode.Combine()```, which was introduced in 2.1

Reply:
This confused me as well - what's the use case? At the very least, can we rename it something like HashNameAndTags ? A negative (valueless) method name is usually harder to infer meaning.

Reply:
yea, I'll rename it and add some comments since this seems like a confusing part.